### PR TITLE
pushable_object

### PR DESCRIPTION
**Pull request by kakaomonk**

### CODE REVIEW

**Comment by kakaomonk** [comment]
I fixed the major error of pushable element. Pushablelements can be pushed now. 
TODO: Implement if statement when there are concatenated PushableElements and When it's beside the wall.

File: src/main/java/game/Game.java [hunk 1]
```diff
@@ -37,9 +39,27 @@ public void movePlayer(Point2D move) {
...
+            if (object instanceof PushableElement) {
+                pushables.add(object);
+            }
+        }
 
         if (traversable)  {
             player.setPos(newPos);
+            for (Element pushable : pushables) {
+                if (pushable.getPos().getX() == player.getPos().getX() &
+                        pushable.getPos().getY() == player.getPos().getY()) {
```
Comment (gwengou) [inline src/main/java/game/Game.java:55]: pushable.getPos() and player.getPos() can be compared using Point2D.equals() method.

Comment (kakaomonk) [reply src/main/java/game/Game.java:55]: Oh yeah. Thanks for your suggestion! I will make a change.

Comment (kakaomonk) [reply src/main/java/game/Game.java:55]: > 

I think we should stick with comparing the value of the X and Y coordinates. .equals() does not comparing the value of X,Y coordinates.

Comment (gwengou) [reply src/main/java/game/Game.java:55]: ? It does. 

'''
public static boolean equals(Point2D p1, Point2D p2) {
        return (p1.getX() == p2.getX()) & (p1.getY() == p2.getY());
    }
'''

**Comment by AditMeh** [comment]
I tried it out. The pushable goes inside the wall. Is this normal?

**Comment by kakaomonk** [comment]
> I tried it out. The pushable goes inside the wall. Can you make it so that both the player and pushable don't move the pushable is next to a wall?

Yes, that's what I'm working now

**Comment by kakaomonk** [comment]
I close this pull request, and we will pull "revised_pushable" branch, we've used the same arithmetic logic.

**Pull request closed without merging**